src/web/utils/textHelpers.spec.ts

```diff
@@ -216,4 +217,15 @@ describe('Text helper tests', () => {
       ]);
     });
   });
+
+  it('should check if string is alphanumeric with hyphen and dot', () => {
```

Comment:
Looks like you also have a helper method for underscores.  should add a test for that here, too.